druid-filtering-ext/README.md

```diff
@@ -0,0 +1,151 @@
+## Druid Filtering Ext
+*This repository contains different filtering extensions*
+
+## Introduction
+
+Apache Druid is a high-performance, column-oriented, distributed data store designed for fast analytics on large datasets.
+One of the powerful features of Druid is its filtering capabilities, which allow users to limit the data returned by queries based on specific criteria.
+
+This repository includes three custom filters designed to enhance IP address filtering capabilities within Druid:
+
+1. **SingleTypeIPRangeFilter (`type = ip_single_range`)**
+2. **MultiRangeIPFilter (`type = ip_multi_range`)**
+3. **FixedSetIPFilter (`type = ip_fixed_range`)**
+
+These filters enable users to filter records based on IP address ranges and fixed sets of IPs, supporting both IPv4 and IPv6 formats.
+
+---
+
+## Filters Overview
+
+### 1. SingleTypeIPRangeFilter
+
+**Description**:
+`SingleTypeIPRangeFilter` allows filtering of records based on a single range of IP addresses of the same type (either IPv4 or IPv6).
+It simplifies the process of retrieving data relevant to a specific IP range.
+
+**Usage**:
+This filter is particularly useful in scenarios where data needs to be analyzed or aggregated based on specific IP ranges, 
+such as monitoring traffic from a particular geographical location or service.
+
+**Example**:
+```json
+{
+  "type": "ip_single_range",
+  "dimension": "ipColumn",
+  "range": {
+    "lower": "192.168.1.1",
+    "upper": "192.168.1.100",
+    "lowerOpen": false,
+    "upperOpen": false
+  },
+  "ignoreVersionMismatch": true
+}
+```
+
+#### Parameter Descriptions
```

Comment:
Parameter names are a bit confusing i suggest smth like
lower -> ipRangeLowerBound
upper -> ipRangeUpperBound
lowerOpen -> isLowerBoundOpen
upperOpen -> isUpperBoundOpen
ignoreVersionMismatch -> ignoreIPVersionMismatch

Reply:
It is something similar to `range` filter. I think we added it as part of design. Let me know we still need to change it.

Reply:
I would make field names more informative @cmick WDYT?